### PR TITLE
feat: update to hash-to-curve-07

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: src/bls12_381/fq.rs [hunk 1]
```diff
@@ -552,37 +551,15 @@ impl BaseFromRO for Fq {
...
-            0xdcff7fffffffd555u64,
-            0x0f55ffff58a9ffffu64,
-            0xb39869507b587b12u64,
-            0xb23ba5c279c2895fu64,
-            0x258dd3db21a5d66bu64,
-            0x0d0088f51cbff34du64,
-        ]);
-
-        if self.into_repr().cmp(&PM1DIV2) == Ordering::Greater {
+        if self.into_repr().0[0] & 1 == 1 {
```
Comment (nikkolasg) [inline src/bls12_381/fq.rs:555]: Are you sure the array is always of length at least 1 ? And is there no way to use the same style of equality as before ?

File: src/signum.rs [hunk 2]
```diff
@@ -12,6 +12,31 @@ pub enum Sgn0Result {
...
+    }
+}
+
+impl BitAnd for Sgn0Result {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        if self == rhs && self == Sgn0Result::Negative {
+            // 1 & 1 == 1
+            Sgn0Result::Negative
```
Comment (nikkolasg) [inline src/signum.rs:33]: I trust it is working but the logic here evades me: this is saying that a Negative AND NonNegative = NonNegative, right ?
Isn't that weird ? Or do we only say a Fq2 is negative when both Fq are negative ? Might be, i'm just asking questions here from curiosity.

**Review by nikkolasg — APPROVED** [review]
It's LGTM for me as the test vectors pass, although the logic around the sign is a bit cryptic to me !

File: src/bls12_381/fq.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/bls12_381/fq.rs:555]: Yes `fq`s representation is a fixed size array of 6 elements. 
This is the check the spec recommends, so I am using that (x mod 2)

File: src/signum.rs (hunk 2 above)
Comment (dignifiedquire) [reply src/signum.rs:33]: This should follow the logic from the rfc, which says that `negative = 1` and `non negative = 0`. Based on that the logical and rules are applied for `1` and `0` here.

**Pull request merged**